toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,227 @@
+	    popup.addSeparator();
+	    popup.add(executeItem);
+	    popup.addSeparator();
+	    popup.add(settingsItem);
+	    popup.addSeparator();
+	    popup.add(quitItem); 
+	    
+	    return popup;
 	}
 	
-	ActionListener getKillListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	agentServer.unRegister();
-	        	System.exit(-1);
-	        }
-	    };
-	    return listener;
+	private void connectListener(ActionEvent e){
+		connect();
 	}
-	
-	ActionListener getStartListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	try {
-	        		if(connectedToWebApp) {
-	        			boolean flag = false;
-		        		
-						if(verificationWebApp(chromeDriverName)) {
-							if(verificationWebApp(recorderName)) {
-								if(verificationWebApp(webAppName)) {
-									flag = true;
-									listenerStarted = true;
-									browserManager.startRecording();
-								}
-							}
-						}
-						
-						if(!flag) {
-							NotificationManager.showMessage("Unable to start recorder, please check recoder parameters !").showNotification();
-						}
-	        		}
-	        		else 
-						NotificationManager.showMessage("You have to be connected to the WebApp !").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
+
+	private void connect() {
+		try {
+			if(isWebAppListening()) {
+				if(agentServer.register(webConfigProvider.get().getApiKey())) {
+					makeAgentOnline();
+					NotificationManager.showMessage("Web Agent - Connected to Webapp !").showNotification();
+				}
+				else{
+					NotificationManager.showMessage("The ApiKey does not match with the WebApp: \n" + webConfigProvider.get().getApiKey()).showNotification();
 				}
-	        }
-	    };
-	    return listener;
+			}
+			else {
+				NotificationManager.showMessage("The Web App does not anwser").showNotification();
+			}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
 	}
-	
-	ActionListener getStopListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	if(listenerStarted) {
-		        	browserManager.closeBrowser();
-	        	}
-	        	else
-					NotificationManager.showMessage("The recorder has not been started !").showNotification();
-	        }
-	    };
-	    return listener;
+
+	private void makeAgentOnline() {
+		trayIcon.setImage(onlineImage);
+		connectedToWebApp = true;
 	}
-	
-	ActionListener getSettingsListener(){
-	    ActionListener listener = new ActionListener() {
-	        public void actionPerformed(ActionEvent e) {
-	        	try {
-					new ConfigPanel(webProperties, toastWebPropertiesFile);
-				} catch (IOException e1) {
-					e1.printStackTrace();
-				}
-	        }
-	    };
-	    return listener;
+
+	private boolean isWebAppListening() throws IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)